data/campaigns/Descent_Into_Darkness/scenarios/07a_A_Small_Favor.cfg

```diff
@@ -752,7 +752,9 @@
         [floating_text]
             x,y=$x1,$y1
             color=255,0,0
+            # wmllint: markcheck off
             text=-1
```

Comment:
wmllint considers `text=` a translatable key. However, here we've got a numerical value, so is there any language that needs to have numbers translated? If so, the correct fix would be marking this string as translatable, otherwise I'd probably need to make wmllint skip the translation mark check for strings made only of numerical characters.

Reply:
> is there any language that needs to have numbers translated?

Yes, I think so. Any language using a non-Latin script will probably need to translate numbers. That said, putting them in the po file is probably the wrong answer even then – the correct answer would be to use localization routines to format the number (which will also insert commas or the equivalent as needed).

I was actually thinking of adding an `amount=` key to `[floating_text]` to enable that treatment.

Reply:
I'd guess `+1`, will work in most languages, including the non-Latin ones, based on looking at the translations of ANL's `Farmers produce +1 gold`. A few are converting the number to text, but that looks like a translator's choice.

However, for the ones that translate it to text, they probably need to know the type of object that it refers to, so that they can use the right gender (etc).

Arabic would likely add a right-to-left text marker.

Reply:
I think converting the number to text is simply incorrect – translators shouldn't be doing that.

Anything that doesn't involve rendering the numbers as text can be handled by numeric localization instead of gettext. Hence, my proposal to add `amount=` which formats the number according to the active locale (using the correct digit set, potentially adding commas or the like, etc) and uses that as the text.

Reply:
I'm not seeing the use-case for `amount=`. For all the languages that Wesnoth supports,  I think the "correct digit set" is Arabic Numerals, and I'd be surprised if any language uses commas for the range of numbers that are likely to appear in floating texts. Assuming it will be hitpoints or gold, it's unlikely to be large enough for a separator.

The edge case would be right-to-left languages, where we need to know which direction to display a two-digit number in. @hermestrismi, do you have any comments on that please?

Reply:
But the thing is that using numeric localization handles all of those edge-cases for us. Sure, it's unlikely that the amount will be more than 999, but by using numeric localization, we ensure it's correct even if it _does_ get that large (which could happen in something like Legend of the Invincibles, for example). By using numeric localization, I _think_ we'd ensure that right-to-left languages are also handled correctly. Sure, all current languages use arabic numerals, but if we use the numeric localization system, it will just automatically use the correct alternate digit set if some other new language is added.

Basically, all your arguments in that post boil down to "we don't need it because it makes no difference in our limited usage case". But it's still correct even in our limited usage case, and if we _don't_ do it, we're potentially going to be in a situation someday where it's incorrect because we didn't do it. I'd even go so far to argue that _all_ numbers ever displayed by the engine should pass through numeric localization, which might be kind of a chore since they're probably scattered throughout the code… searching for `std::to_string` would probably find most of them, though.

Reply:
Either way, it sounds like this is "correct enough" as it is, and the other solutions mentioned should be their own issue/PR.